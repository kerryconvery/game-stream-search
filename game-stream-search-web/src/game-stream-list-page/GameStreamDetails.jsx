import React from 'react';
import { string, bool, number } from 'prop-types';
import { makeStyles } from '@material-ui/core/styles';
import Avatar from '@material-ui/core/Avatar';
import Tooltip from '@material-ui/core/Tooltip';

const useStyles = makeStyles(theme => ({
  root: {
    'font-family': 'Helvetica',
    color: '#231E1D',
  },
  streamTitle: {
    display: '-webkit-box',
    '-webkit-box-orient': 'vertical',
    '-webkit-line-clamp': '2',
    fontSize: '16px',
    textOverflow: 'ellipsis',
    overflow: 'hidden',
    whiteSpace: 'normal',
    marginTop: 0,
    marginBottom: '0.25rem',
  },
  link: {
    textDecoration: 'none',
    color: 'inherit',
  },
  detailsContainer: {
    display: 'flex',
    flexDirection: 'row',
    paddingTop: '1rem',
  },
  streamDetails: {
    paddingLeft: '0.5rem',
  },
  streamSubDetails: {
    color: '#606060',
    fontSize: '14px',
  },
}));

const GameStreamDetails = ({
  streamTitle,
  streamThumbnailUrl,
  streamUrl,
  streamerName,
  streamerAvatarUrl,
  platformName,
  viewCount }) => {

  const classes = useStyles();

  return (
    <div className={classes.root}>
      <a href={streamUrl} target='_blank' className={classes.link}>
        <img src={streamThumbnailUrl} width={320} height={180} />
        <div className={classes.detailsContainer}>
          <Avatar src={streamerAvatarUrl} />
          <div className={classes.streamDetails}>
            <Tooltip title={streamTitle}>
              <h3 className={classes.streamTitle}>{streamTitle}</h3>
            </Tooltip>
            <div className={classes.streamSubDetails}>
              <div>{streamerName}</div>
              <span>{platformName}</span>
              <span> â€¢ </span>
              <span>{`${viewCount} viewers`}</span>
            </div>
          </div>
        </div>
      </a>
    </div>
  )
}

GameStreamDetails.propTypes = {
  streamTitle: string.isRequired,
  streamerName: string.isRequired,
  streamThumbnailUrl: string.isRequired,
  streamUrl: string.isRequired,
  streamerAvatarUrl: string.isRequired,
  platformName: string.isRequired,
  viewCount: number.isRequired,
}

export default GameStreamDetails;