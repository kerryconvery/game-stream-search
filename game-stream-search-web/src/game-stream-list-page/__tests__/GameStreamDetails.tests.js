import React from 'react';
import { render } from '@testing-library/react';
import GameStreamDetails from '../GameStreamDetails';

describe('Game Stream Details', () => {
  it('should match the snapshot', () => {
    const stream = {
      streamTitle: 'test stream',
      streamerName: 'test streamer',
      streamThumbnailUrl: 'http://test.stream.thumbnail.url',
      streamUrl: 'http://test.stream.url',
      channelThumbnailUrl: 'http://test.channel.thumbnail.url',
      platformName: 'test platform',
      viewCount: 100
    }

    const { container } = render(<GameStreamDetails {...stream}  />);

    expect(container.firstChild).toMatchSnapshot();
  })
})
