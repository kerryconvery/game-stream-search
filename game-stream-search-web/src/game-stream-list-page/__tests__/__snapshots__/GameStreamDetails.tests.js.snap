// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Game Stream Details should match the snapshot 1`] = `
<div
  class="makeStyles-root-1"
>
  <a
    class="makeStyles-link-3"
    href="http://test.stream.url"
    target="_blank"
  >
    <img
      height="180"
      src="http://test.stream.thumbnail.url"
      width="320"
    />
    <div
      class="makeStyles-detailsContainer-4"
    >
      <div
        class="MuiAvatar-root MuiAvatar-circle"
      >
        <img
          class="MuiAvatar-img"
          src="http://test.channel.thumbnail.url"
        />
      </div>
      <div
        class="makeStyles-streamDetails-5"
      >
        <h3
          class="makeStyles-streamTitle-2"
          title="test stream"
        >
          test stream
        </h3>
        <div
          class="makeStyles-streamSubDetails-6"
        >
          <div>
            test streamer
          </div>
          <span>
            test platform
          </span>
          <span>
             â€¢ 
          </span>
          <span>
            100 viewers
          </span>
        </div>
      </div>
    </div>
  </a>
</div>
`;
